### PR TITLE
adding some ifeval guidance

### PR DESCRIPTION
**Pull request by kalexand-rh**
Since we are discussing the way we use if statements to vary context, it seems like a good time to add the historic team knowledge to the contributors guide.

### CODE REVIEW

**Review by kmccarron-rh — COMMENTED** [review]
Looks good! - Just a minor nit

**Review by mikemckiernan — COMMENTED** [review]
One comment.

Separately, I have an interest in learning whether it is a preference or a technical reason to use `ifeval` in modules to set attributes based on the context rather than having assemblies set the attribute and consume the module.

**Review by bergerhoffer — COMMENTED** [review]
Just two minor suggestions!

**Comment by bergerhoffer** [comment]
/remove-lifecycle stale

@kalexand-rh was there anything major holding up adding this? I know that this was something that would be useful to get in (we shared it with the Telco team when giving an overview of conditional usage)

**Comment by jboxman-rh** [comment]
> Since we are discussing the way we use if statements to vary context, it seems like a good time to add the historic team knowledge to the contributors guide.

I didn't realize it was historical knowledge that `ifeval` was banned when used in conjunction with versions? When did this happen these past 3 years?

Thanks!

File: contributing_to_docs/doc_guidelines.adoc [hunk 1]
```diff
@@ -457,6 +457,72 @@ If it makes more sense in context to refer to the major version of the product i
...
+\endif::restricted[]
+
+...
+
+\ifeval::["{context}" == "updating-restricted-network-cluster"]
+:!restricted:
+\endif::[]
+----
+
+Note that you must set and unset each temporary attribute that you introduce to an assembly. Use the temporary attributes in the applicable ifdef and ifndef statements to vary text between the assemblies. The preceeding example uses `restricted` as the temporary attribute to display an additional paragraph for the assembly with the `updating-restricted-network-cluster` context attribute.
```
Comment (kalexand-rh) [inline contributing_to_docs/doc_guidelines.adoc:524]: @bergerhoffer, is this clearer?

Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:524]: @kalexand-rh sorry for the delay. Yes this is much clearer thanks!

Comment (kalexand-rh) [reply contributing_to_docs/doc_guidelines.adoc:524]: @bergerhoffer, I lost track of it for longer. :) Thank you!

**Pull request merged**